capi/geos_ts_c.cpp

```diff
@@ -6266,7 +6266,7 @@ extern "C" {
                             const Geometry* p)
     {
 
-        double length;
+        double length = NAN;
```

Comment:
We recently got rid of the NAN macros, so if NAN is useful here maybe use `std::numeric_limits<double>::quiet_NaN()` or whatever it is? I don't think the default value is actually important because I don't think GEOSLength is actually capable of producing an error.

Reply:
This isn't using `getLength()`, it's using `GEOSLength_r` which can produce errors if the handle or the geometry are NULL.

I've used NAN here because length is then used as a denominator in a function. The other option would be to check the output of `GEOSLength_r` and return an error somehow in `GEOSProjectNormalized_r`, `GEOSInterpolateNormalized_r`; but I don't know what would a proper value be in that case (0?, NAN?).

Reply:
Yes, better to check the return of `GEOSLength_r` and propagate the error value of `GEOSProject_r` / `GEOSInterpolateNormalized_r`.

Reply:
Looking at the code it seems that the sane thing to do would be to return -1 in `GEOSProjectNormalized_r` (the return error for GEOSProject_r) and 0 in `GEOSInterpolateNormalized_r` (the return error for GEOSInterpolate_r), right?

Reply:
Yes, that's what I was trying to say.